test/fixtures/cookbooks/test/recipes/unstable_chef_server.rb

```diff
@@ -0,0 +1,17 @@
+#
+# This recipe tests the :unstable channel.
+# This channel is only accessible from Chef's internal
+# network and contains untested builds of Chef products.
+#
+# Make sure you set below environment variables for
+# tests to work correctly.
+#
+
+ENV['ARTIFACTORY_USERNAME'] = node['artifactory']['username']
+ENV['ARTIFACTORY_PASSWORD'] = node['artifactory']['password']
+
+chef_ingredient 'chef-server' do
+  action :install
+  channel :unstable
+  version :latest
```

Comment:
It feels like we should expose `artifactory_username` and `artifactory_password` attributes on the `chef_ingredient` resource. These should be part of the public API of this resource and it's safer than multiple layers of environment variables.